cmake/SpectreLoadMetadata.cmake

```diff
@@ -0,0 +1,57 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Read the following information from the `Metadata.yaml` file:
+# - SPECTRE_NAME: Name
+# - SPECTRE_VERSION: Version
+# - SPECTRE_DOI: Doi
+# - SPECTRE_ZENODO_ID: ZenodoId
+
+file(READ "${CMAKE_SOURCE_DIR}/Metadata.yaml" SPECTRE_METADATA_FILE)
+# Split lines into cmake list (see https://cmake.org/pipermail/cmake/2007-May/014222.html)
```

Comment:
[optional] I don't think you need to cite the mailing list. We do this sort of thing in several other locations in the code already